llama_index/core/base_retriever.py

```diff
@@ -72,7 +73,29 @@ def _retrieve_from_object(
                 f"Retrieving from object {obj.__class__.__name__} with query {query_bundle.query_str}\n",
                 color="llama_pink",
             )
-        if isinstance(obj, NodeWithScore):
+
+        if isinstance(obj, str):
+            return [
+                NodeWithScore(
+                    node=TextNode(text=obj),
+                    score=score,
+                )
+            ]
+        elif isinstance(obj, dict):
```

Comment:
when would this be true? shouldn't we assume that any linked object has been "deserialized" into its original form already? otherwise this is hard to reason about

Reply:
Actually this is the main entry point

`node.dict()` will serialize both the node and object

But when deserializing, the node remains a dict, and we parse that here back into a node

Reply:
oh i was thinking it would be deserialized when the IndexNode is reconstructed in the vector store

What if IndexNode overrode `from_dict` (or whatever the equivalent is), and handled deseralization in there instead?

Reply:
Yea, we could handle it there

Reply:
done :)